src/main/java/com/ncedu/fooddelivery/api/v1/repos/order/OrderRepo.java

```diff
@@ -39,4 +40,14 @@
             nativeQuery = true
     )
     List<Order> getOrdersByCourierId(@Param(value = "id") Long courierId);
+
+    @Query(value = "SELECT * FROM orders " +
+                   "WHERE courier_id = :courierId AND " +
+                   "      date_start >= :startTime " +
+                   "      AND (cast(cast(:endTime as text) as timestamp) IS NULL OR date_start <= cast(cast(:endTime " +
+                   "      as  text) as timestamp))",
+                   nativeQuery = true)
```

Comment:
Мне кажется, что ты перемудрил. Можно же в функцию, если endTime == null, передавать LocalDateTime.now() и тогда все схлопнется в удобоваримую конструкцию!
```suggestion
    @Query(value = "SELECT * FROM orders " +
                   "WHERE courier_id = :courierId AND " +
                   "      date_start >= :startTime " +
                   "      AND date_start <= :endTime",
                   nativeQuery = true)
```

Reply:
Супер, я просто не додумался :/